### ISSUE TITLE
Can not reproduce the experiment result

### ISSUE THREAD

**Issue by HaoYang-219** [peustr/A2SNN#7]
Hello, I make some mistakes when implement your code.
I try to reproduce the experiment result in Table 1, most of the codes are extracted from your github link. I use optimizer Adam, learning rate 0.01,  and the loss I choose the loss=loss_func(logits, target) - torch.log(wca) - torch.log(me)  ​as train_stochastic do. The model architecture are the same as your code, the settings​ are follows config/cifar10/m5.json,  but I have not found your test algorithm FGSMMC,PGDMC,BIMMC,CWMC in your foolbox, so I use attack PGD and FGSM as L2P's codes. but the accuracy in pgd attack I reproduce is 50+, and the accuracy in fgsm attack is 60+, which bothered me for days, Can you tell me the reason? Thanks a lot.
​

**Comment by peustr** [comment]
Hi,

This repository is under active maintenance, which means that the code that reproduces the experiments in the paper may be slightly different than the current version, because changes have been made.

Moreover, in follow-up research we found that the maximum entropy regularizer is relatively unstable when combined with anisotropic noise. So we are taking steps to eliminate it altogether and focus on the WCA term.

So - unless you are in the mood for extensive tuning - please feel free to use our code as a guide to run your own experiments rather than try to replicate the exact numbers.

**Comment by HaoYang-219** [comment]
Hi,
I have some doubts while reading  your paper.
 First, The Table I you use the backbone ResNet-18, However, the L2P and above results use the backbone ResNet18-V2. The comparison seems unreasonable.

Second, For PGD attack, you use alpha=eps/10 and steps k=10, however, In PNI, the settings are not the same. For example, K=7 in their setting, So why you use these settings and if you do not use these settings attack, could you provide your final attack setting?

At last, The Figure 1 no defense curve in A2SNN is quite different from L2P's Figure 2 and Figure 3 when suffers the same eps attack, since you use almost the same backbone and it's no defense, the results should similar.

Hopes for your reply, Thanks!

**Comment by peustr** [comment]
Hi,

1) If you take a close look at the codebase of L2P you will see that RN-18 and RN-18-V2 are the same architecture.
2) You may change the number of iterations [here](https://github.com/peustr/A2SNN/blob/master/test.py#L9), to make sure the comparisons are fair.
3) They are not too dissimilar, are they? What you are looking for is the trend. The exact numbers do not matter as much, because they can vary due to a number of things, such us the learning rate schedule, the weight decay etc.
